tests/integration/test_permissions_api.py

```diff
@@ -196,11 +196,13 @@ def test_add_permissions_when_permission_already_exist_returns_400(
         ]),
     )
 
-    assert resp.status_code == 400
-    assert resp.json['message'] == 'Permission already exist.'
+    assert resp.status_code == 201
+
+    resp = client.get('/deposits/', headers=auth_headers_for_superuser)
+    assert owner.email in resp.json['hits']['hits'][0]['access']['deposit-read']['users']
 
 
-def test_add_permissions_when_permission_doesnt_exist_returns_400(
+def test_add_permissions_when_permission_doesnt_exist_returns_201(
```

Comment:
Let's add a test with submitting multiple permissions with one failing to check `.rollback`